src/execute.rs

```diff
@@ -205,3 +211,17 @@ fn get_owner(deps: Deps, env: Env, position: u16) -> Option<String> {
         Err(_) => None,
     }
 }
+
+fn check_sufficient_funds(funds: Vec<Coin>, required: &Coin) -> Result<(), ContractError> {
+    if required.amount.u128() == 0 {
+        return Ok(());
+    }
+    let sent_sufficient_funds = funds
+        .iter()
+        .any(|coin| coin.denom == required.denom && coin.amount.u128() >= required.amount.u128());
```

Comment:
this checks for incorrect denom, so i think we should add a test for incorrect denom

Reply:
Will do, but I dont think we can test cw20 send since we dont have the hook messages in place. Therefore it's not possible to send cw20 tokens for minting.

Reply:
In that case, don't worry about that. Just incorrect denom would work